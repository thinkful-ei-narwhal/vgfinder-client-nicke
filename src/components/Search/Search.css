h1.smaller {
  padding: 0;
  margin: 0;
  margin-top: 10px;
  font-size: 30px;
}

.genre {
  font-weight: bold;
}

.padder {
  padding-right: 10px;
}

.sidebyside {
  margin-top: 20px;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.sidebyside-input {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.fontawesome-sizing {
  font-size: 25px;
  text-align: center;
  vertical-align: middle;
}

@media only screen and (max-width: 600px) {
  .sidebyside {
    flex-direction: column;
  }
  .sidebyside-input {
    display: inline;
    padding-left: 30px;
  }
}
