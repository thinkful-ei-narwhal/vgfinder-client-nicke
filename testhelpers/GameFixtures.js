const GameFixtures = {
  stubGameData: {
    id: "1",
    title: "Hollow Knight",
    description:
      "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
    genre: "Adventure",
    rating: "4",
    developer: "Team Cherry",
    trailer_url:
      "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
    image_url_box_art:
      "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
    image_url_two:
      "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
    image_url_three:
      "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
  },
  stubGameListData: [
    {
      id: "1",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
    {
      id: "2",
      title: "Hollow Knight",
      description:
        "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
      genre: "Adventure",
      rating: "4",
      developer: "Team Cherry",
      trailer_url:
        "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
      image_url_box_art:
        "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      image_url_two:
        "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
      image_url_three:
        "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
    },
  ],
};

export default GameFixtures;
