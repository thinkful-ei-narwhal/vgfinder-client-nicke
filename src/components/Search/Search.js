import React, { Component } from 'react';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import GamesList from '../../components/GamesList/GamesList';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Input, Section } from './../../components/Utils/Utils';
import './Search.css';

export default class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      searchVal: '',
      genreVal: "All"
    }
  }

  handleGenreControl = (event) => {
    this.setState({ genreVal: event.target.value });
  }

  handleSearchControl = (event) => {
    this.setState({ searchVal: event.target.value });
  }

  render() {
    let gamesList = this.props.gamesList;

    if (this.state.genreVal !== "All") {
      gamesList = gamesList.filter(game => game.genre === this.state.genreVal);
    }
    const lowerCaseVal = this.state.searchVal.toLowerCase();
    const sortedList = gamesList.filter(game => game.title.toLowerCase().startsWith(lowerCaseVal));
    sortedList.sort(function compare(a, b) {
      return a.title.toLowerCase() > b.title.toLowerCase() ? 1 : -1;
    })

    return (
      <>
        <div className='genre'>
          <label htmlFor='search_genre'>Genre</label>
          <select required id='search_genre' name='search_genre' value={this.state.genreVal} onChange={(e) => this.handleGenreControl(e)}>
            <option value="All">All</option>
            <option value="Action">Action</option>
            <option value="Adventure">Adventure</option>
            <option value="Simulation">Simulation</option>
            <option value="Strategy">Strategy</option>
            <option value="RPG">RPG</option>
            <option value="Sports">Sports</option>
            <option value="Horror">Horror</option>
            <option value="FPS">FPS</option>
            <option value="Third Person">Third Person</option>
            <option value="Point And Click">Point And Click</option>
            <option value="Mystery">Mystery</option>
            <option value="RTS">RTS</option>
            <option value="Puzzle">Puzzle</option>
          </select>
        </div>

        <div>
          <label htmlFor='Search'>
            <FontAwesomeIcon icon={faSearch} />
          </label>
          <Input name='Search' id='Search' placeholder="Search" value={this.state.searchVal} onChange={(e) => this.handleSearchControl(e)} ></Input>
        </div>

        <Section>
          <GamesList gamesList={sortedList} />
        </Section>
      </>
    )
  }
}