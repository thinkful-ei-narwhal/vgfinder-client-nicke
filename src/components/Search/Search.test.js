import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Search from "./Search";
import GameFixtures from "./../../../testhelpers/GameFixtures";

describe(`Search component`, () => {
  it("renders a Search component with supplied props", () => {
    const wrapper = shallow(
      <Search gamesList={GameFixtures.stubGameListData} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
