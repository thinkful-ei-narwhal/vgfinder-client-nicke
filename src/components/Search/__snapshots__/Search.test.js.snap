// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders a Search component with supplied props 1`] = `
<Fragment>
  <Section
    className="search-container"
  >
    <div
      className="sidebyside"
    >
      <div
        className="genre"
      >
        <label
          className="padder genre"
          htmlFor="search_genre"
        >
          Genre
        </label>
        <Select
          id="search_genre"
          name="search_genre"
          onChange={[Function]}
          required={true}
          value="All"
        >
          <Option
            value="All"
          >
            All
          </Option>
          <Option
            value="Action"
          >
            Action
          </Option>
          <Option
            value="Adventure"
          >
            Adventure
          </Option>
          <Option
            value="Simulation"
          >
            Simulation
          </Option>
          <Option
            value="Strategy"
          >
            Strategy
          </Option>
          <Option
            value="RPG"
          >
            RPG
          </Option>
          <Option
            value="Sports"
          >
            Sports
          </Option>
          <Option
            value="Horror"
          >
            Horror
          </Option>
          <Option
            value="FPS"
          >
            FPS
          </Option>
          <Option
            value="Third Person"
          >
            Third Person
          </Option>
          <Option
            value="Point And Click"
          >
            Point And Click
          </Option>
          <Option
            value="Mystery"
          >
            Mystery
          </Option>
          <Option
            value="RTS"
          >
            RTS
          </Option>
          <Option
            value="Puzzle"
          >
            Puzzle
          </Option>
        </Select>
      </div>
      <div
        className="sidebyside-input"
      >
        <label
          className="padder"
          htmlFor="Search"
        >
          <FontAwesomeIcon
            border={false}
            className="fontawesome-sizing"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f002",
                  "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                ],
                "iconName": "search",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </label>
        <Input
          id="Search"
          name="Search"
          onChange={[Function]}
          placeholder="Search"
          value=""
        />
      </div>
    </div>
  </Section>
  <Section>
    <GamesList
      gamesList={
        Array [
          Object {
            "description": "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
            "developer": "Team Cherry",
            "genre": "Adventure",
            "id": "2",
            "image_url_box_art": "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
            "image_url_three": "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
            "image_url_two": "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
            "rating": "4",
            "release_date": "2018-07-22",
            "title": "Hollow Knight",
            "trailer_url": "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
          },
          Object {
            "description": "Forge your own path in Hollow Knight! An epic action adventure through a vast ruined kingdom of insects and heroes. Explore twisting caverns, battle tainted creatures and befriend bizarre bugs, all in a classic, hand-drawn 2D style.",
            "developer": "Team Cherry",
            "genre": "Adventure",
            "id": "1",
            "image_url_box_art": "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
            "image_url_three": "https://i.pinimg.com/originals/35/b6/1e/35b61ecc72dc845891b779b6e6ce79bd.jpg",
            "image_url_two": "https://hollowknight.com/wp-content/uploads/2018/09/false_knight.jpg",
            "rating": "4",
            "release_date": "2018-07-22",
            "title": "Hollow Knight",
            "trailer_url": "https://www.youtube.com/watch?v=UAO2urG23S4&ab_channel=TeamCherry",
          },
        ]
      }
    />
  </Section>
</Fragment>
`;
